### ISSUE TITLE
Ability for user to use any avatar

### ISSUE THREAD

**Issue by philchuang** [JabbR/JabbR#724]
My company's firewall blocks gravatar - so it would be nice to have other avatar options, such as the ability to set any image URL.

I suggest a command like 
/avatar https://si0.twimg.com/profile_images/3072076506/5217356076347c3d6ca87f1e46909e99_bigger.jpeg

**Comment by davidfowl** [comment]
@philchuang sounds like a good pull request :smile:

**Comment by philchuang** [comment]
I don't know enough about the architecture to see how deeply entrenched gravatar is, so one day i'll dive in, take a look and let you know :)

**Comment by philchuang** [comment]
Looks like ChatUser.Hash property is being used exclusively for gravatar.

When the user uses the /gravatar command and provides an email address, that email address is md5-hashed and then stored in the ChatUser.Hash property.

In several web pages then, the hash is used to construct a gravatar image url.

So I'm thinking, change ChatUser.Hash to ChatUser.ImageUrl - have the gravatar url be generated completely on the server side. Then the web pages just use the image url as it is and does not have to do any string manipulation.

So we could simply do an /avatar [image_url] command which would work alongside the existing /gravatar [email] command.

Would this be the correct way to do it?

**Comment by davidfowl** [comment]
@philchuang I would add another field that just took the url and then have the client use that url if the hash is null.

**Comment by 0vermind** [comment]
+1 for this
@davidfowl are you working on this? Otherwise I can try to do it.

**Comment by davidfowl** [comment]
Nope I'm not working on it. It should be pretty easy to do. Please checkout the guidelines https://github.com/JabbR/JabbR/blob/master/CONTRIBUTING.md

**Comment by philchuang** [comment]
I'm still undecided about the suggested implementation. I don't like the idea of having the avatar url being handled  separately twice (in the client) - one way for the gravatar, and another way for an arbitrary avatar.

**Comment by davidfowl** [comment]
You don't have to have 2 things. The url is the url and by default it would be a gravatar url.

**Comment by samandmoore** [comment]
Yeah. Just store the gravatar URL as the default and allow people to
override.
